Tests/SonarQube.Bootstrapper.Tests/BuildAgentUpdaterTests.cs

```diff
@@ -68,6 +70,96 @@ public void Updater_CheckDownloadUrl_ConnectionFailure()
+                    string path = Path.Combine(destinationDir, BuildAgentUpdater.LoaderTargetsName);
+                    File.Delete(path);
+                }
+            }
+        }
+
+        [TestMethod]
+        [Description("The targets should be copied if they don't exist. If they have been changed, the updater should overwrite them")]
+        public void InjectTargetsFile_Overwrite()
+        {
+            // In case the dummy targets file somehow does not get deleted (e.g. when debugging), make sure its content valid XML
+            string sourceTargetsContent1 = @"<Project ToolsVersion=""4.0"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"" />";
+            string sourceTargetsContent2 = @"<Project ToolsVersion=""12.0"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"" />";
+
+            CreateDummySourceTargetsFile(sourceTargetsContent1);
+
+            try
+            {
+                InjectTargetsFileAndAssert(sourceTargetsContent1, expectCopy: true);
+                Assert.IsTrue(BuildAgentUpdater.DestinationDirs.Count >= 2, "Expecting at least 2 destination directories");
+
+                string path = Path.Combine(BuildAgentUpdater.DestinationDirs[0], BuildAgentUpdater.LoaderTargetsName);
+                File.Delete(path);
+
+                CreateDummySourceTargetsFile(sourceTargetsContent2);
+                InjectTargetsFileAndAssert(sourceTargetsContent2, expectCopy: true);
+
+            }
+            finally
+            {
+                foreach (string destinationDir in BuildAgentUpdater.DestinationDirs)
+                {
+                    string path = Path.Combine(destinationDir, BuildAgentUpdater.LoaderTargetsName);
+                    File.Delete(path);
+                }
+            }
+        }
+
+        private static void CreateDummySourceTargetsFile(string sourceTargetsContent1)
+        {
+            string bootstrapperLocation = Path.GetDirectoryName(typeof(BuildAgentUpdater).Assembly.Location);
+            string dummyLoaderTargets = Path.Combine(bootstrapperLocation, BuildAgentUpdater.LoaderTargetsName);
+
+            if (File.Exists(dummyLoaderTargets))
+            {
+                File.Delete(dummyLoaderTargets);
+            }
+
+            File.AppendAllText(dummyLoaderTargets, sourceTargetsContent1);
+        }
+
+        private static void InjectTargetsFileAndAssert(string sourceTargetsContent, bool expectCopy)
+        {
+            BuildAgentUpdater updater = new BuildAgentUpdater();
+            TestLogger logger = new TestLogger();
+            updater.InjectLoaderTargets(logger);
+
+            foreach (string destinationDir in BuildAgentUpdater.DestinationDirs)
+            {
+                string path = Path.Combine(destinationDir, BuildAgentUpdater.LoaderTargetsName);
+                Assert.IsTrue(File.Exists(path), "Targets file not found at: " + path);
+                Assert.AreEqual(sourceTargetsContent, File.ReadAllText(path));
```

Comment:
A higher-level assertion message would be better: "Target does not have the expected content. Path: {0}".